notation.go

```diff
@@ -240,6 +251,7 @@ func Verify(ctx context.Context, verifier Verifier, repo registry.Repository, re
 	}
 
 	var verificationOutcomes []*VerificationOutcome
+	var failedOutcomes []*VerificationOutcome
```

Comment:
Why does failedOutcomes needs to be an array? Can we use use a boolean value to check if verification was successful but metadata match failed?

Reply:
i guess it doesn't necessarily need to be an array. my reasoning was that it might be easier to prioritize different error messages based on different failure outcomes in the future (not just a metadata failure), but this can also just be a single boolean/message that's set if the outcome error is metadata verification failure.

Reply:
Since we dont have check on number of signatures user can verify adding an unbounded array will aggravate the unbounded memory consumption issue.